its/src/test/java/com/sonar/it/scanner/msbuild/sonarqube/SslTest.java

```diff
@@ -0,0 +1,173 @@
+import com.sonar.it.scanner.msbuild.utils.TestUtils;
+import com.sonar.orchestrator.build.BuildFailureException;
+import com.sonar.orchestrator.build.BuildResult;
+import com.sonar.orchestrator.locator.FileLocation;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.io.TempDir;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.sonarqube.ws.Issues;
+
+import static com.sonar.it.scanner.msbuild.sonarqube.Tests.ORCHESTRATOR;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@ExtendWith(Tests.class)
+public class SslTest {
+  private static final Logger LOG = LoggerFactory.getLogger(SslTest.class);
+  private static final String PROJECT_KEY = "ssl-certificate";
+  private static final String SSL_KEYSTORE_PASSWORD_ENV = "SSL_KEYSTORE_PASSWORD";
+  private static final String SSL_KEYSTORE_PATH_ENV = "SSL_KEYSTORE_PATH";
+
+  private static String keystorePath;
+  private static String keystorePassword;
+
+  @TempDir
+  public Path basePath;
+
+  @BeforeAll
+  public static void init() {
+    keystorePath = System.getenv(SSL_KEYSTORE_PATH_ENV);
+
+    if (keystorePath == null) {
+      LOG.error("Missing environment variable: " + SSL_KEYSTORE_PATH_ENV);
+      throw new IllegalStateException("Missing environment variable: " + SSL_KEYSTORE_PATH_ENV);
+    }
+
+    keystorePassword = System.getenv(SSL_KEYSTORE_PASSWORD_ENV);
+
+    if (keystorePassword == null) {
+      LOG.error("Missing environment variable: " + SSL_KEYSTORE_PASSWORD_ENV);
+      throw new IllegalStateException("Missing environment variable: " + SSL_KEYSTORE_PASSWORD_ENV);
+    }
+  }
+
+  @BeforeEach
+  public void setUp() {
+    TestUtils.reset(ORCHESTRATOR);
+  }
+
+  /**
+   * Test SSL connection to SonarQube server while the certificate is trusted in the system store.
+   * <p>
+   * To run this test locally, it is required to set the following environment variables:
+   * <ul>
+   *   <li>SSL_KEYSTORE_PATH: path to the keystore file</li>
+   *   <li>SSL_KEYSTORE_PASSWORD: password of the keystore file</li>
+   * </ul>
+   * <p>
+   * The keystore file should be a PKCS12 file containing the certificate and the private key.
+   * Prior to running the test, the keystore should be added to the Windows ROOT store:
+   * <pre>
+   *   certutil -f -p password -importPFX path-to-keystore
+   * </pre>
+   *
+   * The <code>scripts\generate-and-trust-self-signed-certificate.ps1</code> script can be used to:
+   * <ul>
+   *   <li>generate the self-signed certificate</li>
+   *   <li>add it to the system truststore</li>
+   *   <li>set the environment variables</li>
+   * </ul>
```

Comment:
Consider placing this comment on the init() method, because init() will fail if the certificate is not there. You can point from here to the init() method with a comment.